datadog_checks_base/datadog_checks/base/utils/db/statement_metrics.py

```diff
@@ -104,6 +113,47 @@ def compute_derivative_rows(self, rows, metrics, key):
         return result
 
 
+def merge_duplicate_rows(rows, metrics, key):
+    """
+    Given a list of query rows, apply query normalization and compute query signatures to detect duplicate
+    queries_by_key and merge them into a single row.
+
+    - **rows** (_List[dict]_) - rows from current check run
+    - **metrics** (_List[str]_) - the metrics to compute for each row
+    - **key** (_callable_) - function for an ID which uniquely identifies a query row across runs
+    """
+
+    queries_by_key = OrderedDict()
+    for row in rows:
```

Comment:
We have to loop over rows for the merging operation here but we could skip one loop if we just inlined that logic in the existing loop in `compute_derivative_rows`. 

It does feel a little cleaner to have the merge be done in its separate function but is that the right call?

Reply:
Having two O(n) loops is perfectly acceptable. We're iterating 10K rows at _most_ once every 10-15s, so doing multiple passes is negligible.